File: prolog/ignisia.pl
/* Places on Ignisia */
on(ignisia, ignisia_old_mine).
on(ignisia, ignisia_northern_volcano).
on(ignisia, ignisia_western_volcano).
on(ignisia, ignisia_eastern_volcano).

entry(ignisia_old_mine).
port(ignisia, ignisia_old_mine).

/* Descriptions of locations on Ignisia */
describe(ignisia) :- write('You are on Ignisia, a volcanic planet. It''s incredibly hot here.'), nl.

describe(ignisia_old_mine) :- write('You are at the old mine. It''s the only stable ground around.'), nl.
describe(ignisia_northern_volcano) :- write('You are at the northern volcano.'), nl.
describe(ignisia_western_volcano) :- write('You are at the western volcano.'), nl.
describe(ignisia_eastern_volcano) :- write('You are at the eastern volcano.'), nl.
describe(ignisia_mine_control_room) :- write('You are inside the mine control room. The intense heat makes you sweat.'), nl.

/* Items on Ignisia */
at(data_fragment_pyro, ignisia_mine_control_room).
at(encrypted_door, ignisia_mine_control_room).

/* Navigation on Ignisia */
path(ignisia_old_mine, n, ignisia_northern_volcano).
path(ignisia_old_mine, w, ignisia_western_volcano).
path(ignisia_old_mine, e, ignisia_eastern_volcano).

path(ignisia_northern_volcano, s, ignisia_old_mine).
path(ignisia_western_volcano, e, ignisia_old_mine).
path(ignisia_eastern_volcano, w, ignisia_old_mine).

path(ignisia_eastern_volcano, s, ignisia_mine_control_room).

path(ignisia_mine_control_room, n, ignisia_eastern_volcano).

File: haskell/app/Ignisia.hs
module Ignisia where

import Cordinations
import Items
import Doors

places :: [(Planet, String)]
places =
    [ (Ignisia, "ignisia_old_mine"),
      (Ignisia, "ignisia_northern_volcano"),
      (Ignisia, "ignisia_western_volcano"),
      (Ignisia, "ignisia_eastern_volcano"),
      (Ignisia, "ignisia_mine_control_room")
    ]

items :: [ItemPlacement]
items =
    [ (DataFragmentPyro, (Ignisia, "ignisia_mine_control_room"))
    ]

doors :: [DoorPlacement]
doors =
    [ (EncryptedDoor, (Ignisia, "ignisia_eastern_volcano"), South)
    ]

extraLocationMessages :: [(Place, [String])]
extraLocationMessages =
    [ ( (Ignisia, "ignisia_old_mine"),
        [ "The ground radiates intense heat. This old mine is the only stable spot.",
          ""
        ]
      ),
      ( (Ignisia, "ignisia_eastern_volcano"),
        [ "A path along the volcano’s edge leads to a hidden control room inside.",
          ""
        ]
      ),
      ( (Ignisia, "ignisia_mine_control_room"),
        [ "The heat is overwhelming, and a glowing data fragment rests in the center.",
          ""
        ]
      )
    ]

move :: Place -> Direction -> Place
move (Ignisia, "ignisia_old_mine") North = (Ignisia, "ignisia_northern_volcano")
move (Ignisia, "ignisia_old_mine") West  = (Ignisia, "ignisia_western_volcano")
move (Ignisia, "ignisia_old_mine") East  = (Ignisia, "ignisia_eastern_volcano")
move (Ignisia, "ignisia_northern_volcano") South = (Ignisia, "ignisia_old_mine")
move (Ignisia, "ignisia_western_volcano") East   = (Ignisia, "ignisia_old_mine")
move (Ignisia, "ignisia_eastern_volcano") West   = (Ignisia, "ignisia_old_mine")
move (Ignisia, "ignisia_eastern_volcano") South  = (Ignisia, "ignisia_mine_control_room")
move (Ignisia, "ignisia_mine_control_room") North = (Ignisia, "ignisia_eastern_volcano")
move (planet, location) _ = (planet, location)
